src/System.Net.Sockets/tests/FunctionalTests/CreateSocketTests.netcoreapp.cs

```diff
@@ -0,0 +1,49 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.IO;
+using System.IO.Pipes;
+using System.Threading.Tasks;
+using Microsoft.DotNet.RemoteExecutor;
+using Xunit;
+
+namespace System.Net.Sockets.Tests
+{
+    public partial class CreateSocket
+    {
+        [Theory]
+        [InlineData(AddressFamily.Netlink)]
+        [InlineData(AddressFamily.Packet)]
+        [InlineData(AddressFamily.ControllerAreaNetwork)]
+        [PlatformSpecific(~TestPlatforms.Linux)]
+        public void Ctor_Netcoreapp_Throws(AddressFamily addressFamily)
+        {
+            // All protocols are Linux specific and throw throw on other platforms
+            Assert.Throws<SocketException>(() => new Socket(addressFamily, SocketType.Raw, 0));
+        }
+
+        [Theory]
+        [InlineData(AddressFamily.Netlink)]
+        [InlineData(AddressFamily.Packet)]
+        [InlineData(AddressFamily.ControllerAreaNetwork)]
+        [PlatformSpecific(TestPlatforms.Linux)]
+        public void Ctor_Netcoreapp_Success(AddressFamily addressFamily)
+        {
+            try
+            {
+                Socket s = new Socket(addressFamily, SocketType.Raw, 0);
+                s.Close();
```

Comment:
Can you move the Close out of the try block?

Reply:
Close() would be valid only of socket creation succeed, right? Or do you want me to do s?.Close() and move `Socket s` up?

Reply:
```C#
Socket s = null;
try
{
    s = new Socket(...);
}
catch (SocketException e) when (...)
{
    return;
}
s.Close();
```